components-java-sdk/src/main/java/io/dapr/components/domain/bindings/InputBinding.java

```diff
@@ -28,9 +27,10 @@ public interface InputBinding extends InitializableWithProperties, Pingable {
    * will close the stream and return the status on any error. In case of closed
    * connection, the client should re-establish the stream.
    *
-   * @param request A Flux representing the initial request object and the stream of
-   *                acknowledgements sent back by the client to the component.
+   * @param first The initial request object.
+   * @param acks A flux representing stream of acknowledgements sent back by the client
+   *            to the component.
    * @return A Flux containing the responses returned by the component.
    */
-  Mono<ReadResponse> read(Flux<ReadRequest> request);
+  Flux<ReadResponse> read(ReadRequest first, Flux<ReadRequest> acks);
```

Comment:
Just a small first step towards addressing #26

Reply:
Is there a reason why the first request is split from the remainders?  In the case of pub-sub, it was because the first request specified the topic while the remainders indicated acks of messages.  I didn't think it was quite the same pattern for input bindings; in that case, I thought all requests were acks.

Reply:
You are probably right. I had the impression that the first message for InputBinding#read was also special in that sense. Let me just confirm on the Go code.

Reply:
Fixed in c2118d079950bbca990d1cb54e59826edb1d9ac9